### ISSUE TITLE
Building EPSG:4326 tiles

### ISSUE THREAD

**Issue by mmacaula** [gbif/openmaptiles#3]
@MattBlissett, I came across your fork looking to build my own set of tiles for the 4326 projection.  I've been trying to do this for my project for several days with the base OpenMapTIles and haven't gotten very far.  Then I found this fork!  Thank you for your work! 

I'm very interested in doing this for my project as well, but noticed some references to internal images, (`mb.gbif.org:5000/import-osm:latest`, `mb.gbif.org:5000/import-osmborder`, etc).  Are those images or their source for building published anywhere?  I imagine they are modified versions of those tools to handle the projection.  I'm really only interested in generating the tiles for now.  Any general direction you could point me to get this working outside the gbif org?

EDIT: I found the repos for the other tools.  https://github.com/gbif/import-osm for example.  I will continue pressing on, if you have any other advice though I'd still appreciate it.

**Comment by MattBlissett** [comment]
(Back from travels/holiday now.)

Hi,

It's a bit of a rough project, as once it ran correctly I moved on with other work.  If you'd like the mbtiles file for testing I can put it somewhere, it's 68GB.

It is nearing time I refreshed our tilesets, as they are now 5 years old (!).  I came across planetiler and wondered if that might be a better way: https://github.com/onthegomap/planetiler/discussions/284

And I also think targetting EPSG:4087 projection (units are metres rather than degrees) might be easier, as it will mean fewer changes needed to the layer definitions.

**Comment by mmacaula** [comment]
> If you'd like the mbtiles file for testing I can put it somewhere, it's 68GB. 

I appreciate the offer but we're hoping to get current data.

>I came across planetiler and wondered if that might be a better way 

Seems like it might not be ready for that but I did see they were in talks to merge planettiler into openmaptiles.  

I haven't made a lot of progress directly generating the tiles myself.  Instead I'm doing a pretty convoluted route of using mapproxy to take the existing mbtiles and reproject them.  I'd love to simplify that if I could.  It seems openmaptiles has updated their tooling and their layer defs a lot since you forked.  Seems the layer defs also don't have (from what I can tell) explicit references to units (like metres) anymore either.  I confess I'm fairly new to this geospatial stack but under some guidance be able to put in some time to get it up and going again.  

From reading your fork it appears the following happened last time: 

1. You modified iposm to import into 4326 so you could import the OSM data appropriately.  I'm not sure how exactly you did  that with the other projections
2. You modified the layer definition SQL to use degrees before, but that's different now.  It seems they're using a `ZRES*` or `Z(int)` set of functions which might simplify things if we could redefine them?  I couldn't locate where those were defined.  
3. I think you also switched to using tilestache to generate the tiles as tilelive did not support the 4326 projection 
4. Did you also write your own servers to serve up the different projections?

That's my poor take on your work.

**Comment by karatemir** [comment]
@mmacaula do you have any updates on building your own 4326 tiles?

**Comment by mmacaula** [comment]
No, I haven't had time to work on anything directly and was waiting to see if @MattBlissett could confirm or update my summary of his work before I tried.

**Comment by MattBlissett** [comment]
[island_rank.sql](https://github.com/openmaptiles/openmaptiles/blob/master/layers/place/island_rank.sql) -- this is a case where the units are relevant. Those figures are in square metres, but WGS84 calculations would be in square degrees.

1. Yes. Once [import-osm](https://github.com/gbif/import-osm) imported to 4326, I converted from that to the polar projections using SQL within PostGIS, e.g. https://github.com/gbif/openmaptiles/blob/master/layers/landcover/landcover_3575.sql , except for water which used https://github.com/gbif/gbif-basemaps/tree/master/polar-water-tiles
2. Yes. I have a note that `zres(zoom) → 40075016.6855785/((1.0*pixel_scale)*2**zoom)`, and DB dump with 
```
CREATE FUNCTION zres(z double precision) RETURNS double precision
    LANGUAGE plpgsql IMMUTABLE
    AS $$
begin
    return (40075016.6855785/(256*2^z));
end;
$$;

CREATE FUNCTION z(numeric) RETURNS integer
    LANGUAGE plpgsql IMMUTABLE
    AS $_$
begin
    -- Don't bother if the scale is larger than ~zoom level 0
    if $1 > 600000000 then
        return null;
    end if;
    return round(log(2,559082264.028/$1));
end;
$_$;
```
although I'm not sure that's the final DB.

3. I used a modified TileStache at first, but switched to [postserve](https://github.com/gbif/postserve) to generate the vector tiles directly in PostGIS.  There are Python servers in there.

The whole thing was run with a Python script to generate messages (one per tile) and dump them in RabbitMQ (which we have running anyway). The resulting tiles were put on another queue, and another script would save them to a MBTiles DB.  All these scripts are in here: https://github.com/gbif/openmaptiles/tree/master/bin
